crates/ol/stf/src/transaction_processing.rs

```diff
@@ -175,7 +181,12 @@ fn apply_interactions<S: IStateAccessor>(
     fx_buf: AcctInteractionBuffer,
     context: &BasicExecContext<'_>,
 ) -> ExecResult<()> {
-    // Send the messages off to each of the targets.
+    // Process transfers: pure value transfers with no message data
+    for t in fx_buf.transfers {
+        account_processing::process_transfer(state, source, t.dest, t.value, context)?;
+    }
+
+    // Process messages: carry both value and data
     for m in fx_buf.messages {
         account_processing::process_message(state, source, m.dest, m.payload, context)?;
     }
```

Comment:
Are all of these `t`s and `m`s processing atomic?
What happens if for example `t_1` goes through without errors, but `t_2` errors? `m_1` will never be processed right even though `t_1` was processed?